src/main/java/pl/javastart/zadanie21/ProductController.java

```diff
@@ -0,0 +1,51 @@
+package pl.javastart.zadanie21;
+
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+
+import java.util.Set;
+
+@Controller
+public class ProductController {
+
+    private final ProductRepository productRepository;
+
+    public ProductController(ProductRepository productRepository) {
+        this.productRepository = productRepository;
+    }
+
+    @GetMapping("/lista")
+    public String list(@RequestParam(value = "kategoria", required = false) ProductCategory category, Model model) {
+        Set<Product> products;
+        if (category != null) {
+            products = productRepository.findByCategory(category);
+            model.addAttribute("category", category.getTranslation());
+        } else {
+            products = productRepository.findAll();
+            model.addAttribute("category", "wszystkie");
+        }
+        model.addAttribute("products", products);
+
+        double sum = 0;
+        for (Product product : products) {
+            sum += product.getPrice();
+        }
```

Comment:
W kontrolerze nie powinno być żadnej logiki, nawet takiego prostego sumowania. Zamiast tego lepiej byłoby umieścić odpowiednią metodę w repozytorium, albo wprowadzenie dodatkowej warstwy pośredniej, czyli np. klasę ProductService, która korzystałaby z repozytorium, a kontroler korzystałby wtedy z ProductService, a nie bezpośrednio z repozytorium.
Ten fragment kodu mógłby wyglądać tak:
```
double sum = productRepository.sumProducts(category);
```
Jeżeli category byłoby nullem, to sumowane byłyby wszystkie produkty